App.js

```diff
@@ -1,69 +1,182 @@
+        tokenEndpoint: `${authUrl}/oauth/token`
+    }
+};
 
 // a StackNavgiator will give the ability to "push a screen"
 // for instance, when a user clicks a event cell it will push a detailed view on the stack
 const ScheduleStack = createStackNavigator({
-  Schedule,
-  Event
-})
+    Schedule,
+    Event
+});
 
 const HomeStack = createStackNavigator({
-  Home
-})
+    Home
+});
 
 const WorkshopStack = createStackNavigator({
-  Workshops
-})
-
-const TabNavigator = createBottomTabNavigator({
-  Home: HomeStack,
-  Schedule: ScheduleStack,
-  Notifications: WorkshopStack
-}, {
-    defaultNavigationOptions: ({ navigation }) => ({
-      tabBarIcon: ({ focused, horizontal, tintColor }) => {
-        const { routeName } = navigation.state;
-
-        switch(routeName) {
-          case "Home":
-            icon = faHome
-            break;
-          case "Schedule":
-            icon = faList
-            break;
-          case "Notifications":
-            icon = faBell 
-            break;
-          default:
-            icon = null;
-        }
+    Workshops
+});
 
-        return <FontAwesomeIcon color={tintColor} icon={icon} />
-    }
+const LoginStack = createStackNavigator({
+    Login
+});
+
+const TabNavigator = createBottomTabNavigator(
+    {
+        Home: HomeStack,
+        Schedule: ScheduleStack,
+        Notifications: WorkshopStack,
+        Login: LoginStack
+    },
+    {
+        defaultNavigationOptions: ({ navigation }) => ({
+            tabBarIcon: ({ focused, horizontal, tintColor }) => {
+                const { routeName } = navigation.state;
+
+                switch (routeName) {
+                    case "Home":
+                        icon = faHome;
+                        break;
+                    case "Schedule":
+                        icon = faList;
+                        break;
+                    case "Notifications":
+                        icon = faBell;
+                        break;
+                    case "Login":
+                        icon = faKey;
+                        break;
+                    default:
+                        icon = faQuestion;
+                }
 
-    })
-  });
+                return <FontAwesomeIcon color={tintColor} icon={icon} />;
+            }
+        })
+    }
+);
 
 const AppContainer = createAppContainer(TabNavigator);
 
 export default class App extends Component<Props> {
+    state = {
+        user: null,
+        accessToken: null
+    };
+
+    componentDidMount() {
+        AsyncStorage.getItem(
+            "userData",
+            (error, result) =>
+                result && this.setState({ user: JSON.parse(result) })
+        );
+        AsyncStorage.getItem(
+            "accessToken",
+            (error, result) => result && this.setState({ accessToken: result })
+        );
+    }
 
-  render() {
-    return (
-      <AppContainer>Hi</AppContainer>
-    )
-  }
-}
\ No newline at end of file
+    logout = async () => {
```

Comment:
for consistency with react methods, would prefer you make these function declarations over expressions (`function logout() {`). Won't enforce. Difference is not too big